src/resources/ble_darwin.mm

```diff
@@ -318,11 +318,11 @@ explicit BleCentralManagerResource(BleResourceGroup* group)
   void add_discovered_peripheral(DiscoveredPeripheral* discoveredPeripheral) {
     if ([_peripherals objectForKey:[discoveredPeripheral->peripheral() identifier]] == nil) {
       _peripherals[[discoveredPeripheral->peripheral() identifier]] = discoveredPeripheral->peripheral();
-      _newly_discovered_peripherals.append(discoveredPeripheral);
       HostBleEventSource::instance()->on_event(this, kBleDiscovery);
```

Comment:
I think this should be done always after appending.

Reply:
done.